### PR TITLE
Cleaning up a few places in the code

### PR DESCRIPTION
**Pull request by akukanov**
A couple of changes lurking in my working copy, plus the changes for unused variables from #471 that still apply, and dropping colons before `std` in the code around.

### CODE REVIEW

**Comment by SergeyKopienko** [comment]
@akukanov I think technically this patch is OK.
But is it a good practice to mix some significant changes like remove unused param from `__parallel_transform_scan_base` together with some unsignificant changes like remove `::` in one PR?

**Comment by akukanov** [comment]
> @akukanov I think technically this patch is OK. But is it a good practice to mix some significant changes like remove unused param from `__parallel_transform_scan_base` together with some unsignificant changes like remove `::` in one PR?

We have decided in the past that we will gradually remove `::` as we update the code base, Essentially, I removed it from the functions & classes touched by the more significant changes - but yes, it's somewhat wider than only the modified expressions. Since the whole patch is about cleaning up the code without any functionality add or loss, I tend to think it's OK.

File: include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h [hunk 1]
```diff
@@ -21,7 +21,6 @@
 
 #include "../../utils_ranges.h"
 #include "../../iterator_impl.h"
-#include "../../glue_numeric_defs.h"
```
Comment (dmitriy-sobolev) [inline include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h:24]: I suggest including `execution_sycl_defs.h` if we remove `glue_numeric_defs.h`. The later transitively includes  `execution_sycl_defs.h`, which defines `__hetero_tag` and  `device_policy` used in the current file.

Comment (akukanov) [reply include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h:24]: Done.

**Review by dmitriy-sobolev — APPROVED** [review]

**Review by SergeyKopienko — COMMENTED** [review]
LGTM

**Pull request merged**